src/fs.rs

```diff
@@ -0,0 +1,8 @@
+/// Opens a file and returns a useful error if it fails
+/// If possible use this instead of File::open for usability purposes
+pub fn open_file(path: &std::path::Path) -> std::io::Result<std::fs::File> {
```

Comment:
I think you should be able to replace the `&Path` with a generic parameter as in the `File::open` signature:
```rust
pub fn open<P: AsRef<Path>>(path: P) -> Result<File>
```

That should fix the compile error.